infrahub_sdk/store.py

```diff
@@ -20,130 +23,336 @@ def get_schema_name(schema: str | type[SchemaType] | None = None) -> str | None:
     return None
 
 
-class NodeStoreBase:
-    """Internal Store for InfrahubNode objects.
-
-    Often while creating a lot of new objects,
-    we need to save them in order to reuse them later to associate them with another node for example.
-    """
+class NodeStoreBranch:
+    def __init__(self, name: str) -> None:
+        self.branch_name = name
 
-    def __init__(self) -> None:
-        self._store: dict[str, dict] = defaultdict(dict)
-        self._store_by_hfid: dict[str, Any] = defaultdict(dict)
+        self._objs: dict[str, InfrahubNode | InfrahubNodeSync | CoreNode | CoreNodeSync] = {}
+        self._hfids: dict[str, dict[tuple, str]] = defaultdict(dict)
```

Comment:
Do we need this to be a defaultdict or can we just have a `{}` instead?

Reply:
we could do a dict but it means we'll need to add a check to ensure the sub level dict as been initialized
is there a concern using a `defaultdict` ?

Reply:
To me they are just an even less strict data structure than a `dict`, I think the code is often clearer without them.